src/main/java/org/wso2/carbon/esb/connector/MQPublishQueue.java

```diff
@@ -0,0 +1,386 @@
+        mqMessage.correlationId = config.getCorrelationID().getBytes();
+        log.info("Correlation ID " + config.getCorrelationID());
+        log.info("Message ID " + config.getMessageID());
+        if (config.getPriority() != -1000) {
+            mqMessage.priority = config.getPriority();
+        }
+
+        int messageType = config.getMessageType();
+        mqMessage.messageType = messageType;
+
+        if (config.isPersistent()) {
+            mqMessage.persistence = MQPER_PERSISTENT;
+        } else {
+            mqMessage.persistence = MQPER_NOT_PERSISTENT;
+        }
+        if (config.getgroupID() != null) {
+            mqMessage.groupId = config.getgroupID().getBytes();
+        }
+
+        GregorianCalendar cal = new GregorianCalendar();
+        cal.setTime(new Date(System.currentTimeMillis()));
+        mqMessage.putDateTime = cal;
+
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
+        }
+        return mqMessage;
+    }
+
+    /**
+     * Create request MQMesaage
+     */
+    MQMessage buildRequestMessageandGetResponse(MQConfiguration config, String queueMessage) {
+
+        MQMessage mqMessage = new MQMessage();
+
+        mqMessage.messageId = config.getMessageID().getBytes();
+        log.info("message ID " + config.getMessageID());
+        mqMessage.correlationId = config.getCorrelationID().getBytes();
+        log.info("correlation ID " + config.getCorrelationID());
+
+        if (config.getPriority() != -1000) {
+            mqMessage.priority = config.getPriority();
+        }
+
+        int messageType = config.getMessageType();
+        mqMessage.messageType = messageType;
+        String reportQueue = config.getreplyQueue();
+        mqMessage.replyToQueueName = reportQueue;
+
+        if (reportQueue != null) {
+            MQQueue reply;
+            reply = setReplyQueue(this.connectionBuilder, this.config);
+            if (reply != null) {
+                getReplyMessge(reply, this.config);
+            }
+        } else {
+            log.info("Reply queue not specified");
+        }
+        if (config.isPersistent()) {
+            mqMessage.persistence = MQPER_PERSISTENT;
+        } else {
+            mqMessage.persistence = MQPER_NOT_PERSISTENT;
+        }
+        if (config.getgroupID() != null) {
+            mqMessage.groupId = config.getgroupID().getBytes();
+        }
+
+        GregorianCalendar cal = new GregorianCalendar();
+        cal.setTime(new Date(System.currentTimeMillis()));
+        mqMessage.putDateTime = cal;
+        try {
+            mqMessage.writeString(queueMessage);
+        } catch (Exception e) {
+            log.info("Error creating mq message" + e);
+        }
+        return mqMessage;
+    }
+
+    /**
+     * Get reply message
+     */
+    void getReplyMessge(final MQQueue replyQueue, MQConfiguration config) {
```

Comment:
I think can get rid of this method by pointing, ibm-mq inbound to reply queue. Handling Listening part a connector in not good IMO.